File: data/readable_format/fortran/file850.f90
program test_dawrit
    implicit none
    real(8), allocatable :: arr(:)
    integer :: n

    ! Define the array length and allocate it
    n = 5
    allocate(arr(n))
    arr = [1.0, 2.0, 3.0, 4.0, 5.0]

    ! Call the dawrit subroutine
    call dawrit(arr, n, 1)

contains

    subroutine dawrit(V, LEN, NREC)
        integer, intent(in) :: LEN, NREC
        real(8), dimension(LEN), intent(in) :: V
        real(8) :: sum
        integer :: i

        sum = 0.0
        do i = 1, LEN
            sum = sum + V(i)
        end do
        print *, 'Sum = ', sum
    end subroutine dawrit

end program test_dawrit

File: data/readable_format/cpp/file850.cpp
#include <iostream>
#include <vector>

// Define the DawritI class directly in this file
class DawritI {
public:
    static void dawrit(const std::vector<double>& V, int LEN, int NREC);
};

// Implement the dawrit function within the class definition
void DawritI::dawrit(const std::vector<double>& V, int LEN, int NREC) {
    double sum = 0.0;
    for (int i = 0; i < LEN; ++i) {
        sum += V[i];
    }
    std::cout << "Sum = " << sum << std::endl;
}

// The main function to demonstrate usage
int main() {
    std::vector<double> arr = {1.0, 2.0, 3.0, 4.0, 5.0};
    DawritI::dawrit(arr, arr.size(), 1);
    return 0;
}